src/components/navbar/buttons/hamburger.js

```diff
@@ -1,15 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './hamburger.scss';
+import Structure from '../../../structure.json';
+import { IoIosArrowForward } from 'react-icons/io';
+import { IoIosArrowBack } from 'react-icons/io';
+import { GrFormSearch } from 'react-icons/gr';
+
+const Hamburger = ({ showLinks }) => {
+    const [menu] = useState(Structure['menu']);
+    const [showMenu, setShowMenu] = useState(showLinks);
+
+    const [subMenu] = useState(Structure['menu'][5]);
+    const [showSubMenu, setShowSubMenu] = useState(false);
+
+    const controlMenu = () => {
+        setShowMenu(!showMenu);
+        setShowSubMenu(!showSubMenu);
+    }
 
-const Hamburger = () => {
-    
     return (
         <div className='hamburger__links'>
-            <ul className='hamburger__links__ul'>
-                <li><a href="">home</a></li>
-                <li><a href="">shop</a></li>
-                <li><a href="">blog</a></li>
-                <li><a href="">contact</a></li>
+            <ul className={showMenu ? 'hamburger__links__ul hamburger__links__ul--menu' : 'hamburger__links__ul hamburger__links__ul--subMenu'}>
+                {showMenu &&
+                    menu.map((link) => {
+                        if (link.title === "Search") {
+                            return (
+                                <li className='hamburger__links__ul__link hamburger__links__ul__link--search'>
+                                    <input type="text" id="search" name="search" placeholder='Search...' />
+                                    <GrFormSearch className='hamburger__links__ul__link__search' />
+                                </li>
+                            );
+                        } else {
+                            return (
+                                <li className={link.isSubMenu ? "hamburger__links__ul__link hamburger__links__ul__link--isSubMenu" : "hamburger__links__ul__link"}>
+                                    <a href="#">{link.title}</a>
+                                    {link.isSubMenu && <IoIosArrowForward className='hamburger__links__ul__link__arrow' onClick={() => controlMenu()} />}
+                                </li>
+                            );
+                        }
+                    })
+                }
+
+                {showSubMenu &&
+                    <li className="hamburger__links__ul__link hamburger__links__ul__link--subMenuFirst">
+                        <IoIosArrowBack className='hamburger__links__ul__link__arrow' onClick={() => controlMenu()} />
+                        <a href="#" onClick={() => controlMenu()}>back</a>
+                    </li>
+                }
+                {showSubMenu &&
```

Comment:
Why check `showSubMenu` twice?

Reply:
I tried to use one time, but when i used, the first <li> tag did not show up, that is why i used twice. Also, first <li> is not included in json file, therefore, i wrote them separately from map function of second showSubMenu check condition.